monkestation/code/game/objects/items/implants/hardlight.dm

```diff
@@ -31,42 +84,214 @@
+			return . | SPELL_CANCEL_CAST
+
+
+/datum/action/cooldown/spell/conjure_item/hardlight_spear/make_item()
+	. = ..()
+	var/obj/item/gun/magic/hardlight_spear/made_spear = .
+	made_spear.spears_left = spell_level-1
+
+/datum/action/cooldown/spell/conjure_item/hardlight_spear/get_spell_title()
+	switch(spell_level)
+		if(2)
+			return "Upgraded "
+		if(3)
+			return "Recursive "
+		if(4)
+			return "Resonant "
+		if(5)
+			return "Ascended "
+		if(6)
+			return "Overwhelming "
+		if(7)
+			return "Commmanding "
+
+	return ""
+
+/datum/action/cooldown/spell/conjure_item/hardlight_spear/max
+	name = "Commmanding Hardlight Spear"
+	spell_level = 7
+
+/datum/action/cooldown/spell/conjure_item/hardlight_spear/max/get_spell_title()
+	return "" //commanding commanding
+
+/obj/item/gun/magic/hardlight_spear //listen man
+	name = "hardlight spear"
+	desc = "A spear made out of hardened light."
+	fire_sound = 'sound/weapons/fwoosh.ogg'
+	pinless = TRUE
+	force = 25
+	wound_bonus = -5
+	bare_wound_bonus = 20
+	armour_penetration = 18
+	block_chance = 0
+	sharpness = SHARP_POINTY
+	w_class = WEIGHT_CLASS_HUGE
+	antimagic_flags = NONE
+	hitsound = 'sound/weapons/blade1.ogg'
 	icon = 'monkestation/icons/obj/items_and_weapons.dmi'
 	icon_state = "lightspear"
 	inhand_icon_state = "lightspear"
-	icon_prefix = "lightspear"
 	worn_icon_state = "none"
 	lefthand_file = 'monkestation/icons/mob/inhands/polearms_lefthand.dmi'
 	righthand_file = 'monkestation/icons/mob/inhands/polearms_righthand.dmi'
-	name = "hardlight spear"
-	desc = "A spear made out of hardened light."
-	force = 15
-	w_class = WEIGHT_CLASS_BULKY
-	slot_flags = NONE
+	slot_flags = null
+	can_charge = FALSE //ITS A SPEAR
+	item_flags = NEEDS_PERMIT | DROPDEL | ABSTRACT | NO_MAT_REDEMPTION
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 	light_system = MOVABLE_LIGHT
 	light_outer_range = 3
 	light_power = 1
-	throwforce = 25
-	throw_speed = 6
-	armour_penetration = 18
-	hitsound = 'sound/weapons/blade1.ogg'
-	sharpness = SHARP_POINTY
+	max_charges = 1
+	var/spears_left = 5
+	ammo_type = /obj/item/ammo_casing/magic/hardlight_spear
 
-/obj/item/spear/hardlight_spear/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+/obj/item/gun/magic/hardlight_spear/Initialize(mapload)
 	. = ..()
-	if(hit_atom && !QDELETED(hit_atom))
-		if(istype(hit_atom, /mob/living)) //Living mobs handle hit sounds differently.
-			var/volume = get_volume_by_throwforce_and_or_w_class()
-			playsound(hit_atom, 'sound/weapons/genhit.ogg',volume, TRUE, -1)
-		else
-			playsound(src, drop_sound, 60, ignore_walls = FALSE)
-		qdel(src) //Deletes when it gets thrown at somethign
-		return hit_atom.hitby(src, 0, 0, throwingdatum=throwingdatum)
-
-/obj/item/spear/hardlight_spear/unembedded()
+	AddComponent(/datum/component/jousting)
+
+	AddComponent(/datum/component/butchering, \
+		speed = 10 SECONDS, \
+		effectiveness = 70, \
+	)
+
+	block_chance = 5+spears_left*5
```

Comment:
leaving this not as a request but as a reminder incase the block is deemed to bad